File: java/src/main/java/com/kale/interview/repositories/CreatorRepository.java
package com.kale.interview.repositories;

import com.kale.interview.data.Creator;
import org.jooq.DSLContext;
import org.jooq.Record;
import org.jooq.impl.DSL;
import org.springframework.stereotype.Repository;

import java.time.OffsetDateTime;

@Repository
public class CreatorRepository {

    private final DSLContext dsl;

    public CreatorRepository(DSLContext dsl) {
        this.dsl = dsl;
    }

    public Creator findById(String id) {
        Record row = dsl.selectFrom(DSL.table("creators"))
                .where(DSL.field("id", String.class).eq(id))
                .fetchOne();
        return row == null ? null : toCreator(row);
    }

    public void updateBalance(String id, long balanceCents) {
        dsl.update(DSL.table("creators"))
                .set(DSL.field("balance_cents", Long.class), balanceCents)
                .where(DSL.field("id", String.class).eq(id))
                .execute();
    }

    static Creator toCreator(Record row) {
        return new Creator(
                row.get("id", String.class),
                row.get("display_name", String.class),
                row.get("email", String.class),
                row.get("balance_cents", Long.class),
                row.get("created_at", OffsetDateTime.class).toString()
        );
    }
}

File: src/main/kotlin/com/kale/interview/repositories/CreatorRepository.kt
package com.kale.interview.repositories

import com.kale.interview.data.Creator
import org.jooq.DSLContext
import org.jooq.Record
import org.jooq.impl.DSL
import org.springframework.stereotype.Repository
import java.time.OffsetDateTime

@Repository
class CreatorRepository(private val dsl: DSLContext) {

    fun findById(id: String): Creator? {
        return dsl.selectFrom(DSL.table("creators"))
            .where(DSL.field("id", String::class.java).eq(id))
            .fetchOne()
            ?.toCreator()
    }

    fun updateBalance(id: String, balanceCents: Long) {
        dsl.update(DSL.table("creators"))
            .set(DSL.field("balance_cents", Long::class.java), balanceCents)
            .where(DSL.field("id", String::class.java).eq(id))
            .execute()
    }
}

fun Record.toCreator() = Creator(
    id = get("id", String::class.java)!!,
    displayName = get("display_name", String::class.java)!!,
    email = get("email", String::class.java)!!,
    balanceCents = get("balance_cents", Long::class.java)!!,
    createdAt = get("created_at", OffsetDateTime::class.java)!!.toString(),
)
